onnx/backend/test/case/node/lstm.py

```diff
@@ -128,36 +137,128 @@ def export_initial_bias():
+                       [-0.02344924, 0.0769614, -0.10114089],
+                       [0.08732384, -0.13031968, 0.3693238],
+                       [0.09512341, 0.39322734, 0.05026317]]]).astype(np.float32)
+        r = np.array([[[-0.06249112, -0.3027284, -0.2714152, 0.38169748, -0.3848986],
+                       [-0.3015067, -0.11511135, -0.21178144, 0.36077315, -0.15175757],
+                       [-0.3106271, -0.13619286, 0.09167248, -0.00270995, 0.16715324],
+                       [0.3096549, 0.06414723, 0.27308917, -0.36815444, -0.25398546],
+                       [0.40350074, 0.0977748, 0.16108459, -0.39921242, 0.03195441],
+                       [0.32862294, 0.43853426, 0.21549404, 0.43726218, -0.12468177],
+                       [0.17379999, -0.17711008, -0.02272931, 0.15821773, -0.2360906],
+                       [0.04289997, 0.14505583, 0.09238386, 0.15120435, 0.22830069],
+                       [0.03189701, -0.12833786, 0.34228635, 0.17734432, 0.3900627],
+                       [-0.15498602, 0.05439121, -0.33910578, 0.02097705, -0.42195255],
+                       [0.02405897, -0.29173693, 0.28562587, 0.39553738, 0.3302138],
+                       [0.08506113, 0.17034757, 0.39796853, 0.08288926, -0.15668085],
+                       [0.28090197, -0.07625446, -0.1972777, -0.18111584, -0.30605704],
+                       [0.31646246, -0.17583266, -0.30067247, 0.36228096, 0.2409823],
+                       [0.41217238, -0.41253033, 0.4323389, -0.43837905, -0.4471018],
+                       [0.33483976, -0.09845337, -0.41000995, -0.02262691, -0.25162753],
+                       [-0.38541794, -0.16338113, 0.13963765, 0.2420575, 0.01179534],
+                       [0.33970326, 0.16575891, -0.0950945, -0.43333504, -0.12386471],
+                       [0.22664595, 0.3502648, -0.3177863, 0.10857916, -0.06070068],
+                       [0.03233594, 0.15320832, 0.390822, 0.27357525, 0.05000198]]]).astype(np.float32)
+        b_w = np.array([[-0.07438096, 0.0116722, 0.25055438, -0.19189253, -0.31282502,
+                         -0.40339673, 0.41707057, -0.10233101, 0.0879212, 0.41836756,
+                         -0.36332014, -0.41120952, -0.14621562, -0.17775917, -0.06715041,
+                         -0.3470294, -0.2873521, 0.280366, -0.13346279, 0.34218425]]).astype(np.float32)
+        b_r = np.array([[-0.03111535, -0.33301952, -0.17032543, -0.05586442, -0.32023185,
+                         -0.3442515, 0.44716895, -0.2340269, 0.27279443, -0.32590097,
+                         -0.07209095, 0.19784379, 0.27797413, 0.00461715, -0.36984265,
+                         -0.07255584, -0.12558547, -0.11320454, -0.35173178, -0.13761121]]).astype(np.float32)
+        b = np.concatenate((b_w, b_r), axis=1)
+
+        lstm = LSTM_Helper(X=x, W=w, R=r, B=b)
+        output = lstm.step()
+
+        output_precomputed = np.array([[[[-0.09183919, -0.20181324, 0.04475278, -0.12022446,
```

Comment:
Where are these numbers coming from and what is the need to have them here instead of relying on reference implementation? Using precomputed numbers may create unnecessary challenges in future. For instance, If a particular framework generates numbers that do not match these numbers in future, how can we resolve conflicts?

Reply:
It comes from pytorch.
Yes, make sense. I will remove this.